core/src/avm1.rs

```diff
@@ -217,21 +222,19 @@ impl<'gc> Avm1<'gc> {
         }
 
         self.is_reading = true;
-        let current_stack_id = self.stack_frames.len() - 1;
-        let (swf_version, data, pc) = self
-            .stack_frames
-            .last()
-            .map(|frame| (frame.swf_version(), frame.data(), frame.pc()))?;
+        let (frame_cell, swf_version, data, pc) = self.stack_frames.last().map(|frame| {
+            (
+                frame.clone(),
+                frame.read().swf_version(),
+                frame.read().data(),
+                frame.read().pc(),
```

Comment:
`GcCell` is a `Copy` type, no need to clone, we can do `*frame`.

`read` and `write` have a runtime cost, so if possible, try to store it in a var if you are calling it multiple times: `let f = frame.read();`